src/infra/core/tests/tests/repos/test_object_repository_s3.rs

```diff
@@ -171,3 +191,35 @@ async fn test_insert_expect() {
 
     test_object_repository_shared::test_insert_expect(&repo).await;
 }
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#[test_group::group(containerized)]
+#[test_log::test(tokio::test)]
+async fn test_external_urls() {
+    let s3 = LocalS3Server::new().await;
+    let repo = ObjectRepositoryS3Sha3::new(S3Context::from_url(&s3.url).await);
+
+    let hash_foo = Multihash::from_digest_sha3_256(b"foo");
+
+    let expected_external_download_url = {
+        let mut url =
+            Url::parse(&format!("http://127.0.0.1/{TEST_BUCKET_NAME}/{hash_foo}")).unwrap();
```

Comment:
I suspect this won't work on macs where docker/podman run in a VM?

Reply:
Our container runtime should have a method to get the hostname where docker is running on.

Reply:
> I suspect this won't work on macs where docker/podman run in a VM?

At the moment, we are not using docker on macs at CI

> Our container runtime should have a method to get the hostname where docker is running on.

btw, we already have `url` in `LocalS3Server` -- I'll use it